### PR TITLE
Ejm merchant invoice show revenue

### PR DESCRIPTION
**Pull request by EMielke76**

### CODE REVIEW

File: app/controllers/merchant_invoices_controller.rb [hunk 1]
```diff
@@ -7,6 +7,6 @@ def index
   def show
     @merchant = Merchant.find(params[:merchant_id])
     @invoice = Invoice.find(params[:invoice_id])
-    @invoiced_item = InvoiceItem.where(invoice_id: params[:invoice_id])
+    #require "pry"; binding.pry
```
Comment (jaw772) [inline app/controllers/merchant_invoices_controller.rb:10]: delete this in refactor

**Review by jaw772 — APPROVED** [review]
Beautiful work on the potential revenue

**Review by pleonar1 — APPROVED** [review]
nice job! everything looks great

File: app/controllers/merchant_invoices_controller.rb (hunk 1 above)
Comment (LelandCurtis) [inline app/controllers/merchant_invoices_controller.rb:10]: why is there a binding pry here?

File: app/models/invoice.rb [hunk 2]
```diff
@@ -9,4 +9,16 @@ class Invoice < ApplicationRecord
   def customer_name
     (customer.first_name) + " " + (customer.last_name)
   end
+
+  def merchant_items(merchant)
+    Item.joins(:invoice_items).where( items: {merchant_id: merchant.id})
```
Comment (LelandCurtis) [inline app/models/invoice.rb:14]: this doesn't seem to be referencing the current context of invoice. It seems to be calling all items from `Item`. Shouldn't it be `invoice.items.` to scope down to only this invoice's items?

File: app/models/invoice.rb [hunk 3]
```diff
@@ -9,4 +9,16 @@ class Invoice < ApplicationRecord
...
+  end
+
+  def merchant_invoice_items(merchant)
+    InvoiceItem.joins(:item).where( items: {merchant_id: merchant.id})
```
Comment (LelandCurtis) [inline app/models/invoice.rb:18]: should this be `self.invoice_items` not `InvoiceItems`?

**Review by LelandCurtis — APPROVED** [review]
replace `Item` with `self.items

**Pull request merged**